crates/utils/src/settings/structs.rs

```diff
@@ -190,4 +190,12 @@ pub struct SetupConfig {
   pub enable_nsfw: Option<bool>,
   #[default(None)]
   pub community_creation_admin_only: Option<bool>,
+  #[default(None)]
+  pub require_email_verification: Option<bool>,
+  #[default(None)]
+  pub require_application: Option<bool>,
+  #[default(None)]
+  pub application_question: Option<String>,
+  #[default(None)]
+  pub private_instance: Option<bool>,
```

Comment:
Worth emphasizing that even with this setting, other instances can view all posts, comments etc from the instance via federation if it is enabled. And because of that, it  is easy to iterate through all posts and comments using `curl -H 'Accept: application/activity+json' https://ds9.lemmy.ml/post/x`. 

So maybe it should throw an error if both `private_instance` and` federation.enabled` are set (at least until #868 or #1576 are implemented).

Reply:
I can add that check to main.rs